### PR TITLE
Add interfaces support to object analysis

### PR DESCRIPTION
**Pull request by gkoz**

### CODE REVIEW

**Comment by EPashkin** [comment]
Not needed `Interface` case in `rust_type::use_by_ref`?
Same in `needed_upcast::needed_upcast`? Not sure about last.

**Comment by EPashkin** [comment]
Better add some examples in `Gir_Gtk.toml` (as minimum generate `Gtk.Orientable`).

**Comment by gkoz** [comment]
> Not needed Interface case in rust_type::use_by_ref?
Same in needed_upcast::needed_upcast?

You're right about both, updated.

**Comment by EPashkin** [comment]
Thanks.

File: src/analysis/parents.rs [hunk 1]
```diff
@@ -29,3 +29,26 @@ pub fn analyze(env: &Env, type_: &Class, imports: &mut Imports)
     parents.reverse();
     parents
 }
+
+pub fn analyze_interface(env: &Env, type_: &Interface, imports: &mut Imports)
+        -> Vec<StatusedTypeId> {
+    let mut parents = Vec::new();
+
+    for &parent_tid in &type_.prereq_parents {
+        let parent_type = env.type_(parent_tid).to_ref_as::<Class>();
```
Comment (EPashkin) [inline src/analysis/parents.rs]: `Interface`?

**Comment by gkoz** [comment]
@homu r+

**Comment by gkoz** [comment]
@homu r-

**Comment by gkoz** [comment]
I'll look into that one

File: src/analysis/parents.rs (hunk 1 above)
Comment (gkoz) [inline src/analysis/parents.rs]: The prerequisites are usually classes. Not sure if I ever saw an interface depend on other interfaces.

Comment (gkoz) [inline src/analysis/parents.rs]: But yeah, there's a `TreeModel` prerequisite to `TreeSortable` so this needs to be generalized.

**Comment by EPashkin** [comment]
May be both class(`Widget`) and interface(`TreeModel`) prerequisites

**Pull request merged**